### ISSUE TITLE
App Center Data - ability to query partition

### ISSUE THREAD

**Issue by patrickdoodyIRL** [microsoft/appcenter#1252]
The Data API does not provide a way to query data within a partition without first downloading and deserializing the entire contents of a partition. In cases where there are many hundreds of objects within a partition, reading the entire partition requires multiple round trips to page through the available data. 

I am currently caching data locally to cut down on the volume of calls to the Data API, however in doing so we lose the ability to see truly live data.

The Cosmos Client library uses a QueryDefinition class that takes a SQL query string and parameters for filtering data on the server. Given that AppCenter Data requires the backing CosmosDB to use the SQL API, perhaps something similar can be integrated into the Data API?

**Comment by Oddj0b** [comment]
Hi @patrickdoodyIRL, I tagged my colleague and he will look at. Thanks for suggesting the idea.

**Comment by akamud** [comment]
+1 for this.

**Comment by Zakeelm** [comment]
@patrickdoodyIRL   Thank you so much for your request. We announced this week in a [blog post]( https://devblogs.microsoft.com/appcenter/app-center-mbaas-retirement/) that we’re retiring the Auth and Data services later this year and recommending customers transition to  Azure Active Directory B2C and Azure Cosmos DB instead. Let me know if you have any additional questions about this transition and i'll make sure to get them answered! Closing this issue.
